stringbuilder.go

```diff
@@ -336,3 +336,38 @@ func createTrimSet(chars ...rune) map[rune]bool {
 
 	return trimSet
 }
+
+// Set initial position
+func (s *StringBuilder) Rewind() {
+	s.position = 0
+}
+
+// Sets the rune at the specific position
+func (s *StringBuilder) SetRuneAt(index int, val rune) error {
+	if index < 0 {
+		return fmt.Errorf("index should always be greater than or equal to zero")
+	}
+	if index > s.position {
+		return fmt.Errorf("index cannot be greater than current position")
+	}
+	s.data[index] = val
+
+	return nil
+}
+
+// Change current position
+func (s *StringBuilder) Skip(forward int) error {
```

Comment:
Can you give me an example of where this function would be useful?

Imagine we have a string "Hello" and the builder can hold 8 characters. Calling Skip(2) would work but will return a string where the last two characters are either undefined or just empty.